### PR TITLE
Test/refactor target helper and test it

### PR DESCRIPTION
**Pull request by Framonti**
This PR is not associated to any Github issue

* Small refactor of TargetHelper class: change if-else chain with when construct, merged same branches, extract helper method to improve readability
* Introduce tests for TargetHelper, covering most base cases
* Rename  `setTileFeatures` to `setTileATerrainAndFeatures` to better describe what it does
* Add comments to explain how hex grid works

### CODE REVIEW

**Review by yairm210 — APPROVED** [review]

File: core/src/com/unciv/ui/components/UncivTextField.kt [hunk 1]
```diff
@@ -72,7 +72,7 @@ fun TextField.scrollAscendantToTextField(): Boolean {
     }
 
     val scrollContent = scrollPane.actor
-    val textFieldScrollContentCoords = localToAscendantCoordinates(scrollContent, Vector2(0f, 0f))
+    val textFieldScrollContentCoords = localToAscendantCoordinates(scrollContent, Vector2.Zero)
```
[the remark this answers is not in the document]
Comment (yairm210) [reply core/src/com/unciv/ui/components/UncivTextField.kt:75]: Good catch! 0_0

@Framonti This is a blocker

**Comment by yairm210** [comment]
I think the lesson learnt here is to default to 0,0, and only use .zero on places where we've determined it's performance critical
Default to safe

**Comment by Framonti** [comment]
> Missed two: Scene2dExtensions - Actor.stageBoundingBox and FasterUIDevelopment - addBorder

Hi, 
I should have also reverted Actor.stageBoundingBox
In Scene2dExtensions, the vector is only use to initialize a Rectangle using .x and .y values.
If I understand properly, Vector2 uses directly Java primitive type of float, so it should be passed by copy to the constructor call? Feel free to correct me, I'm still learning Kotlin

**Comment by Framonti** [comment]
Wow, such a in-depth and clear response. Many thanks, it was really helpful. 
I'll take much more care to catch the surrounding context of what we're using to avoid this kind of situations in the future

**Pull request merged**